File: data/multilang_comparison/program_go/src/func47.go
package main
import "math/rand"
var _ = rand.Intn
func Func47(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array11 *Array
   if pCounter > 0 {
       array11 = vars.Data[pCounter-1]
       pCounter--
       array11.RefC++
   } else {
       array11 = &Array{
           Size: 919,
           RefC: 1,
           Id: 11,
           Data: make([]uint32, 919),
       }
   }
   return array11;
}

File: data/multilang_comparison/program_go/src/func47.c
#include "program_go.h" 
array_t* func47(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array11;
   if (pCounter > 0) {
      array11 = vars->data[--pCounter];
      array11->refC++;
      DEBUG_COPY(array11->id);
   } else {
      array11 = (array_t*)malloc(sizeof(array_t));
      array11->size = 919;
      array11->refC = 1;
      array11->id = 11;
      array11->data = (unsigned int*)malloc(array11->size*sizeof(unsigned int));
      memset(array11->data, 0, array11->size*sizeof(unsigned int));
      DEBUG_NEW(array11->id);
   }
   return array11;
}

